app/queries/content_dependencies.rb

```diff
@@ -1,40 +1,56 @@
 module Queries
+  # This class resolves the depencies for a given subject content item with
+  # a provided content_id and locale
+  #
+  # There are 3 types of dependency this resolves:
+  # 1 - Content items that are linked to the subject of dependency resolution
+  #     (eg for a subject of a if b has a link to a b will be returned),
+  #     for certain link types these are recursed forming a tree structure.
+  # 2 - Content Items which have an automoatic reverse link to the subject.
+  #     These are items this subject links to and is represented reciprocally
+  #     in the item linked to. eg if our subject (A) has a parent of B, B would
+  #     automatically have a link to A of type children.
+  # 3 - Translations, for the subject all locales except the one provided is
+  #     added as dependencies, as well as all translations of the content_ids
+  #     found.
   class ContentDependencies
-    def initialize(content_id:, fields: [], dependent_lookup: GetDependents.new)
+    def initialize(content_id:, locale:, state_fallback_order:)
       @content_id = content_id
-      @fields = fields
```

Comment:
Yeah that'd be good to reduce the amount it'd need to be worked right across from the request -> worker point. Currently it's redundant as we only pass in the content_id field at a worker point. So I felt it was better to remove it than to leave it as something ambiguous.